UnitySDK/Assets/ML-Agents/Scripts/InferenceBrain/GeneratorImpl.cs

```diff
@@ -109,25 +109,24 @@ public VectorObservationGenerator(ITensorAllocator allocator)
     /// It will use the Memory data contained in the agentInfo to fill the data
     /// of the tensor.
     /// </summary>
-    public class RecurrentInputGenerator : TensorGenerator.Generator
+    public class RecurrentInputGenerator : TensorGenerator.IGenerator
     {
-        private readonly ITensorAllocator _allocator;
+        private readonly ITensorAllocator m_Allocator;
 
         public RecurrentInputGenerator(ITensorAllocator allocator)
         {
-            _allocator = allocator;
+            m_Allocator = allocator;
         }
 
         public void Generate(
             TensorProxy tensorProxy, int batchSize, Dictionary<Agent, AgentInfo> agentInfo)
         {
-            TensorUtils.ResizeTensor(tensorProxy, batchSize, _allocator);
+            TensorUtils.ResizeTensor(tensorProxy, batchSize, m_Allocator);
 
             var memorySize = tensorProxy.shape[tensorProxy.shape.Length - 1];
             var agentIndex = 0;
-            foreach (var agent in agentInfo.Keys)
+            foreach (var memory in agentInfo.Keys.Select(agent => agentInfo[agent].memories))
```

Comment:
I'm still fuzzy on c# - is this idiomatic? faster?

Reply:
seems like the consensus is no.  For now I will remove the linq queries.  I'm not sure whether rider is helping or hurting.